widget/list.go

```diff
@@ -761,6 +761,10 @@ func (l *listLayout) updateList(newOnly bool) {
 		for _, vis := range l.visible {
 			l.setupListItem(vis.item, vis.id, l.list.focused && l.list.currentFocus == vis.id)
 		}
+
+		// a full refresh may change theme, we should drain the pool of unused items instead of refreshing them.
+		for l.itemPool.Get() != nil {
```

Comment:
I don't understand why this is the case. Is it not defeating the purpose of the pool if we do this for each refresh?

Reply:
The pool does not exist to make refresh fast it is to make scroll fast...
When `Refresh()` is called we have to re-draw all of the list items whether they are in the pool or not. So either a) drain the pool and don't refresh them or b) Redraw all the items not currently on screen.

I went with a - but b would be OK too if you have a strong feeling.

Reply:
Makes sense. Let's go with this approach for now